components/p3a/brave_p3a_config.cc

```diff
@@ -0,0 +1,125 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/p3a/brave_p3a_config.h"
+
+#include "base/command_line.h"
+#include "base/logging.h"
+#include "brave/components/p3a/brave_p3a_switches.h"
+#include "brave/components/p3a/buildflags.h"
+
+namespace brave {
+
+namespace {
+
+constexpr uint64_t kDefaultUploadIntervalSeconds = 60;  // 1 minute.
+
+base::TimeDelta GetTimeDeltaFromCommandLine(base::CommandLine* cmdline,
+                                            const char* switch_name) {
+  if (cmdline->HasSwitch(switch_name)) {
+    std::string seconds_str = cmdline->GetSwitchValueASCII(switch_name);
+    int64_t seconds;
+    if (base::StringToInt64(seconds_str, &seconds) && seconds > 0) {
+      return base::Seconds(seconds);
+    }
+  }
+  return base::TimeDelta();
+}
+
+std::string GetStringFromCommandLine(base::CommandLine* cmdline,
+                                     const char* switch_name) {
+  if (cmdline->HasSwitch(switch_name)) {
+    return cmdline->GetSwitchValueASCII(switch_name);
+  }
+  return std::string();
+}
+
+GURL GetURLFromCommandLine(base::CommandLine* cmdline,
+                           const char* switch_name) {
+  if (cmdline->HasSwitch(switch_name)) {
+    GURL url = GURL(cmdline->GetSwitchValueASCII(switch_name));
+#if defined(OFFICIAL_BUILD)
+    CHECK(url.is_valid());
```

Comment:
it's not really clear why CHECKing of cmdline parameters is needed.. I assume this is used for testing, so why kill the browser instead of logging?